src/test/run-pass/zst_enum.rs

```diff
@@ -0,0 +1,21 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![allow(dead_code)]
+
+enum X {
+    A = 42 as isize,
```

Comment:
Can you also add a test that checks whether

```rust
enum A {
     B = 42,
}
enum C {
     D = 42,
     E = A::B as isize,
}
```

errors? That will check whether the const evaluator also produces the correct value.